ucc-demo.ipynb

```diff
@@ -0,0 +1,319 @@
+    "\n",
+    "Starting with `pytket`, let's draw the pre and post compiled-circuits:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Using pytket, show the pre and post compiled basic circuit\n",
+    "import pytket\n",
+    "from pytket.circuit.display import render_circuit_jupyter as Tketdraw\n",
+    "\n",
+    "\n",
+    "tket_circuit = pytket.Circuit(2)\n",
+    "tket_circuit.H(0)\n",
+    "tket_circuit.CX(0, 1)\n",
+    "tket_compiled_circuit = compile(tket_circuit)\n",
+    "\n",
+    "Tketdraw(tket_circuit)\n",
+    "Tketdraw(tket_compiled_circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's do the same using the `cirq` library"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cirq\n",
+    "\n",
+    "qubits = cirq.LineQubit.range(2)\n",
+    "cirq_circuit = cirq.Circuit(\n",
+    "    cirq.H(qubits[0]),\n",
+    "    cirq.CNOT(qubits[0], qubits[1])\n",
+    ")\n",
+    "cirq_compiled_circuit = compile(cirq_circuit)\n",
+    "\n",
+    "print(cirq_circuit)\n",
+    "print(\"------------------------\")\n",
+    "print(cirq_compiled_circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There's a few things to notice from these toy examples:\n",
+    "\n",
+    "* The same `compile` function is used to compile each circuit, even though they are different front-end formats.\n",
+    "* This default `compile` has changed to a target gateset with single qubit rotations and controlled-X gates."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's step it up a little, and throw a more complex circuit at `ucc`, this time using `qiskit` as the front-end:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.circuit.random import random_clifford_circuit\n",
+    "input_gates = [\"cx\", \"cz\", \"cy\", \"swap\", \"x\", \"y\", \"z\", \"s\", \"sdg\", \"h\"]\n",
+    "num_qubits = 10\n",
+    "random_circuit = random_clifford_circuit(\n",
+    "    num_qubits, gates=input_gates, num_gates=10 * num_qubits * num_qubits\n",
+    ")\n",
+    "compiled_random_circuit = compile(random_circuit)\n",
+    "print(f\"Number of multi-qubit gates in original circuit: {random_circuit.num_nonlocal_gates()}\")\n",
+    "print(f\"Gates used in original circuit: {random_circuit.count_ops()}\")\n",
+    "print(f\"Number of multi-qubit gates in compiled circuit: {compiled_random_circuit.num_nonlocal_gates()}\")\n",
+    "print(f\"Gates used in compiled circuit: {compiled_random_circuit.count_ops()}\")"
+   ]
+  },
```

Comment:
This gave error:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 7
      3 num_qubits = 10
      4 random_circuit = random_clifford_circuit(
      5     num_qubits, gates=input_gates, num_gates=10 * num_qubits * num_qubits
      6 )
----> 7 compiled_random_circuit = compile(random_circuit)
      8 print(f"Number of multi-qubit gates in original circuit: {random_circuit.num_nonlocal_gates()}")
      9 print(f"Gates used in original circuit: {random_circuit.count_ops()}")

TypeError: compile() missing required argument 'filename' (pos 2)
```

Reply:
These errors happened again in other cells but will suppress making the same comment